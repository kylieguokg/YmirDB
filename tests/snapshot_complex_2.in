set a 1
set b 1
set g 1
set h 1
list entries
snapshot
set f a b
set e g h
list entries
snapshot
set d f
append e d
list entries
snapshot
rollback 3
snapshot
checkout 4
len a
len b
len d
len e
len f
len g 
len h
forward a
forward b
forward d
forward e
forward f
forward g
forward h
backward a
backward b
backward d
backward e
backward f
backward g
backward h
pluck e 3
push d e
list entries
len a
len b
len d
len e
len f
len g 
len h
forward a
forward b
forward d
forward e
forward f
forward g
forward h
backward a
backward b
backward d
backward e
backward f
backward g
backward h
pop d
append e d
list entries
len a
len b
len d
len e
len f
len g 
len h
forward a
forward b
forward d
forward e
forward f
forward g
forward h
backward a
backward b
backward d
backward e
backward f
backward g
backward h
set k 28
append a k
list entries
max a
max b
max d
max e
max f
max g 
max h
max k
min a
min b
min d
min e
min f
min g 
min h
min k
drop 3
rollback 3
rollback 2
bye
