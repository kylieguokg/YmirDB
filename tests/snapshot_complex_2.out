> ok

> ok

> ok

> ok

> h [1]
g [1]
b [1]
a [1]

> saved as snapshot 1

> ok

> ok

> e [g h]
f [a b]
h [1]
g [1]
b [1]
a [1]

> saved as snapshot 2

> ok

> ok

> d [f]
e [g h d]
f [a b]
h [1]
g [1]
b [1]
a [1]

> saved as snapshot 3

> ok

> saved as snapshot 4

> ok

> 1

> 1

> 2

> 4

> 2

> 1

> 1

> nil

> nil

> a, b, f

> a, b, d, f, g, h

> a, b

> nil

> nil

> d, e, f

> d, e, f

> e

> nil

> d, e

> e

> e

> d

> ok

> d [e f]
e [g h]
f [a b]
h [1]
g [1]
b [1]
a [1]

> 1

> 1

> 4

> 2

> 2

> 1

> 1

> nil

> nil

> a, b, e, f, g, h

> g, h

> a, b

> nil

> nil

> d, f

> d, f

> nil

> d

> d

> d, e

> d, e

> e

> ok

> d [f]
e [g h d]
f [a b]
h [1]
g [1]
b [1]
a [1]

> 1

> 1

> 2

> 4

> 2

> 1

> 1

> nil

> nil

> a, b, f

> a, b, d, f, g, h

> a, b

> nil

> nil

> d, e, f

> d, e, f

> e

> nil

> d, e

> e

> e

> ok

> ok

> k [28]
d [f]
e [g h d]
f [a b]
h [1]
g [1]
b [1]
a [1 k]

> 28

> 1

> 28

> 28

> 28

> 1

> 1

> 28

> 1

> 1

> 1

> 1

> 1

> 1

> 1

> 28

> ok

> no such snapshot

> ok

> bye
